src/pro/javacard/gp/AccessControl.java

```diff
@@ -0,0 +1,769 @@
+package pro.javacard.gp;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.bouncycastle.util.Arrays;
+
+import apdu4j.HexUtils;
+
+/**
+ * Access control Rules implementation (reference document : Secure Element Access Control Version 1.0).
+ *
+ * @author Bertrand Martel
+ */
+public class AccessControl {
+
+	public final static AID ACR_AID = new AID("A00000015141434C00");
+
+	public final static byte ACR_GET_DATA_ALL = 0x40;
+	public final static byte ACR_GET_DATA_NEXT = 0x60;
+
+	private final static byte[] ACR_GET_DATA_RESP = new byte[]{ (byte)0xFF, (byte)0x40 };
+
+	//Access Rule reference data object (p45 Secure Element Access control spec v1.0)
+	private final static byte REF_AR_DO = (byte) 0xE2;
+	private final static byte REF_DO = (byte) 0xE1;
+	private final static byte AID_REF_DO = (byte) 0x4F;
+	private final static byte HASH_REF_DO = (byte) 0xC1;
+
+	private final static byte AR_DO = (byte) 0xE3;
+	private final static byte APDU_AR_DO = (byte) 0xD0;
+	private final static byte NFC_AR_DO = (byte) 0xD1;
+
+	//from Secure Element Access control spec p46, hash length can be 20 (sha1) or 0
+	private final static byte HASH_MAX_LENGTH = (byte) 0x14;
+	private final static byte HASH_MIN_LENGTH = (byte) 0x00;
+
+	//command message data object (p38 Secure Element Access control spec v1.0)
+	private final static byte STORE_AR_DO = (byte) 0xF0;
+	private final static byte DELETE_AR_DO = (byte) 0xF1;
+
+	/**
+	 * Store data status work (p44 Secure Element Access control spec v1.0)
+	 */
+	public final static Map<Integer, String> ACR_STORE_DATA_ERROR = new HashMap<>();
+	static {
+		ACR_STORE_DATA_ERROR.put(0x6381, "Rule successfully stored but an access rule already exists for this target");
+		ACR_STORE_DATA_ERROR.put(0x6581, "Memory problem");
+		ACR_STORE_DATA_ERROR.put(0x6700, "Wrong length in Lc");
```

Comment:
I wish there was something like ISO7816 class in JavaCard, from where this and several other generic constants (and maybe texts) could be re-used from...

Reply:
ok, I will use ISO7816  from apdu4j for the status which are present